solutions/devsprint-attekita-1/GitHubApp/Screens/Components/ButtonView.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  ButtonView.swift
+//  GitHubApp
+//
+//  Created by Pedro Boga on 21/01/22.
+//
+
+import UIKit
+
+struct ButtonViewConfiguration {
+    var title: String
+    var action: UIAction
+}
+
+class ButtonView: UIView {
+    
+    private lazy var button: UIButton = {
+        let button = UIButton()
+        button.layer.cornerRadius = 10
+        button.titleLabel?.textColor = .white
+        button.titleLabel?.font = UIFont.preferredFont(forTextStyle: .headline)
+        button.backgroundColor = .systemBlue
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addSubview(button)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func updateView(with configuration: ButtonViewConfiguration) {
+        button.setTitle(configuration.title, for: .normal)
+        button.addAction(configuration.action, for: .touchUpInside)
```

Comment:
Como o configuration é uma struct, um tipo por valor, a action vai se perder pois não teremos a referência.

O que podemos fazer, é colocar como target um método que teremos na própria ButtonView

button.addAction(#seletor(callAction), for: .touchUpInside)

@objc private func callAction() {
         aqui podemos abrir já o link e colocaríamos o link como string no configurator **Solução mais simples**
       
         ...Ou podemos criar um protocolo delegate para a nossa EmptyView e aqui chamaríamos o delegate exemplo:
         delegate?.didCallAction(self) 
        dessa forma a DetailView implementaria esse delegate e como estamos passando o proprio botão como self no delegate, ela saberia qual botão foi acionado
  }